### PR TITLE
[Feature] lcd rotation

### PR DESCRIPTION
**Pull request by YunSuk-Yeo**
It replace #37.

This PR adds support for the feeder to switch between multiple LCD clients in round-robin fashion in case one becomes unavailable. This becomes especially useful in the context of [multi-party computation](https://en.wikipedia.org/wiki/Secure_multi-party_computation) validator setups (see [horcrux](https://github.com/jackzampolin/horcrux) for Cosmos SDK-based blockchains).

Oracle feeder with multiple LCD clients in an MPC validator setup

Once the feeder notices an LCD client being unavailable, it just talks to the next one on its list, giving the operator enough time for troubleshooting of the failed one.

### CODE REVIEW

File: feeder/src/vote.ts [hunk 1]
```diff
@@ -294,34 +295,66 @@ interface VoteArgs {
...
+
       resetPrevote()
     })
 
     await Bluebird.delay(Math.max(500, 500 - (Date.now() - startTime)))
   }
 }
 
+function rotateLCD(args: VoteArgs, lcdRotate: { client: LCDClient; current: number; max: number }) {
+  if (++lcdRotate.current > lcdRotate.max) {
```
Comment (hanjukim) [inline feeder/src/vote.ts:346]: I think we need to destroy previous LCDClient .. before rotate it. Is there a function?

Comment (YunSuk-Yeo) [reply feeder/src/vote.ts:346]: How to destroy it?
Currently it overrides the previous client.

File: feeder/src/vote.ts [hunk 2]
```diff
@@ -294,34 +295,66 @@ interface VoteArgs {
...
+    lcdRotate.current = 0
+  }
+
+  lcdRotate.client = new LCDClient(buildLCDClientConfig(args, lcdRotate.current))
```
Comment (YunSuk-Yeo) [inline feeder/src/vote.ts:350]: We override client in here

**Comment by gaia** [comment]
This is great, what is the syntax to specify multiple LCDs in order of preference?

How long does it take for it to consider that the current LCD is failed to switch to the next one?

How long does it take for it to check whether the previous one is back online, and switch back to it?

thx!

**Comment by freshfab** [comment]
1. You just add as many `--lcd` flags as you like (see the [feeder README](https://github.com/terra-money/oracle-feeder/tree/main/feeder)). The order of the addresses determines the rotation order.
2. As soon as one LCD client becomes unavailable, the feeder tries to talk to the next one until it finds an available one.
3. It doesn't check whether any previous LCDs are back online. As long as the current LCD is available, it will not switch to another one.

**Comment by gaia** [comment]
ok, one last q: under which condition does it consider an LCD unavailable?  only when it doesn't reply? what if it is sluggish, or replies with a block in the past?

PS: I'm prob asking for too much here, I know ;)

**Pull request merged**